src/utils/temporal.cpp

```diff
@@ -486,21 +491,50 @@ std::pair<DateParameters, LocalTimeParameters> ParseLocalDateTimeParameters(std:
   }
 }
 
-LocalDateTime::LocalDateTime(const int64_t microseconds) {
-  auto chrono_microseconds = std::chrono::microseconds(microseconds);
-  static constexpr int64_t one_day_in_microseconds = std::chrono::microseconds{std::chrono::days{1}}.count();
-  if (microseconds < 0 && (microseconds % one_day_in_microseconds != 0)) {
-    date = Date(microseconds - one_day_in_microseconds);
+LocalDateTime::LocalDateTime(const int64_t microseconds)
+    : us_since_epoch_(std::chrono::sys_time<std::chrono::microseconds>(std::chrono::microseconds(microseconds))) {
+  // Already UTC
+}
+
+LocalDateTime::LocalDateTime(const DateParameters &date_parameters, const LocalTimeParameters &local_time_parameters) {
+  auto us_since_epoch_local = std::chrono::microseconds{Date{date_parameters}.MicrosecondsSinceEpoch() +
+                                                        LocalTime{local_time_parameters}.MicrosecondsSinceEpoch()};
+  const auto *tz = flags::run_time::GetTimezone();
+  if (tz) {
+    // APPLY TIMEZONE (local to UTC)
+    us_since_epoch_ =
+        tz->to_sys(std::chrono::local_time<std::chrono::microseconds>(std::chrono::microseconds(us_since_epoch_local)));
+  } else {
+    // Fallback to UTC
+    us_since_epoch_ = std::chrono::sys_time<std::chrono::microseconds>(std::chrono::microseconds(us_since_epoch_local));
+  }
+}
+
+LocalDateTime::LocalDateTime(const Date &date, const LocalTime &local_time) {
+  auto us_since_epoch_local =
+      std::chrono::microseconds{date.MicrosecondsSinceEpoch() + local_time.MicrosecondsSinceEpoch()};
+  const auto *tz = flags::run_time::GetTimezone();
+  if (tz) {
+    // APPLY TIMEZONE (local to UTC)
+    us_since_epoch_ =
+        tz->to_sys(std::chrono::local_time<std::chrono::microseconds>(std::chrono::microseconds(us_since_epoch_local)));
```

Comment:
Yeah, this is confusing...
First lets get a few things straight:
1. On Linux when you save a timezoned time point, what happens is that your local time gets converted to the equivalent UTC time and that epoch offset is saved. Ex. In Milano now it's 14:30, this will become 12:30 system time.
2. `RETURN timestamp()` returns system time.
3. `RETURN timestamp()`, `RETURN timestamp(LocalDateTime())` and `RETURN timestamp(LocalDateTime(with current date-time))` should all be equivalent. <- This is why I need SysMicrosecondsFromEpoch()
4. Bolt has a LocalDateTime object, but does not support timezones. What they do is they convert the system timestamp to your local time and then get the epoch offset. <- This is why I need MicrosecondsFromEpoch()
5. There are other date-time data types and in order to support them all in the same way, I am using system time (since that is the fixed point).

Reply:
Short answer, yes :D

Reply:
Aha, got it. Thanks for very good explanation!